addons/website_crm_partner_assign/wizard/crm_forward_to_partner.py

```diff
@@ -103,18 +103,18 @@ def action_forward(self):
             self.env['crm.lead'].message_subscribe([partner_id])
         return True
 
-    def get_lead_portal_url(self, lead_id, type):
-        action = type == 'opportunity' and 'action_portal_opportunities' or 'action_portal_leads'
+    def get_lead_portal_url(self, lead):
+        action = lead.type == 'opportunity' and 'action_portal_opportunities' or 'action_portal_leads'
         action_ref = self.env.ref('website_crm_partner_assign.%s' % (action,), False)
         portal_link = "%s/?db=%s#id=%s&action=%s&view_type=form" % (
-            self.env['ir.config_parameter'].sudo().get_param('web.base.url'),
+            lead.get_base_url(),
             self.env.cr.dbname,
-            lead_id,
+            lead.id,
             action_ref and action_ref.id or False)
         return portal_link
 
     def get_portal_url(self):
-        portal_link = "%s/?db=%s" % (self.env['ir.config_parameter'].sudo().get_param('web.base.url'), self.env.cr.dbname)
+        portal_link = "%s/?db=%s" % (self[0].get_base_url(), self.env.cr.dbname)
```

Comment:
ensureone

Reply:
Done, even if I think this is not needed as we always have an object for the template IMO https://github.com/odoo-dev/odoo/blame/ee6146a871267c86484cf524000cc5e20d8adea7/addons/website_crm_partner_assign/data/mail_template_data.xml#L45
That also add back another ICP call